python/cudf/cudf/tests/test_string.py

```diff
@@ -3128,6 +3128,104 @@ def test_string_get_json_object_invalid_JSONPath(json_path):
     with pytest.raises(ValueError):
         gs.str.get_json_object(json_path)
 
+def test_string_get_json_object_allow_single_quotes():
+    gs = cudf.Series(
+        [
+            """
+            {
+                "store":{
+                    "book":[
+                        {
+                            'author':"Nigel Rees",
+                            "title":'Sayings of the Century',
+                            "price":8.95
+                        },
+                        {
+                            "category":"fiction",
+                            "author":"Evelyn Waugh",
+                            'title':"Sword of Honour",
+                            "price":12.99
+                        }
+                    ]
+                }
+            }
+            """
+        ]
+    )
+    assert_eq(
+        gs.str.get_json_object("$.store.book[0].author", allow_single_quotes=True),
+        cudf.Series(["Nigel Rees"], dtype="object"),
+    )
+    assert_eq(
+        gs.str.get_json_object("$.store.book[*].title", allow_single_quotes=True),
+        cudf.Series(["['Sayings of the Century',\"Sword of Honour\"]"], dtype="object"),
```

Comment:
...and similarly below.
```suggestion
        cudf.Series(["['Sayings of the Century',\"Sword of Honour\"]"]),
```